src/pops.cpp

```diff
@@ -45,269 +44,226 @@ bool all_infected(IntegerMatrix susceptible)
+      config.random_seed, config.rows, config.cols, mt,
+      config.latency_period_steps);
+
+  Model<IntegerMatrix, NumericMatrix, int> model(config);
+  for (unsigned current_index = 0;
+       current_index < config.scheduler().get_num_steps(); ++current_index) {
+
+    // if (all_infected(susceptible)) {
+    //   Rcerr << "All suspectible hosts are infected!" << std::endl;
+    //   infected_vector.push_back(Rcpp::clone(infected));
+    //   susceptible_vector.push_back(Rcpp::clone(susceptible));
+    //   resistant_vector.push_back(Rcpp::clone(resistant));
+    //   break;
+    // }
+    IntegerMatrix dispersers(config.rows, config.cols);
+    mortality_tracker_vector.push_back(Rcpp::clone(mortality_tracker));
+    model.run_step(current_index, current_index, infected, susceptible,
+                   total_plants, dispersers, exposed, mortality_tracker_vector,
+                   mortality, temperature, weather_coefficient, treatments,
+                   resistant, outside_dispersers, spreadrate, movements);
+
+    if (config.spread_schedule()[current_index]) {
+      total_dispersers += dispersers;
+    }
+
+    if (config.use_mortality && config.mortality_schedule()[current_index]) {
+      mortality_vector.push_back(Rcpp::clone(mortality));
+    }
+
+    if (config.output_schedule()[current_index]) {
+      infected_vector.push_back(Rcpp::clone(infected));
+      susceptible_vector.push_back(Rcpp::clone(susceptible));
+      resistant_vector.push_back(Rcpp::clone(resistant));
+      total_host_vector.push_back(Rcpp::clone(total_plants));
+      dispersers_vector.push_back(Rcpp::clone(total_dispersers));
+      // exposed_vector = Rcpp::clone(exposed);
+
+      num_infected = sum_of_infected(infected);
+      number_infected.push_back(num_infected);
+      area_infect = area_of_infected(infected, config.ew_res, config.ns_res);
+      area_infected.push_back(area_infect);
+      total_dispersers(config.rows, config.cols);
+    }
+
+    if (config.spread_rate_schedule()[current_index]) {
+      unsigned simulation_year = simulation_step_to_action_step(
+          config.spread_rate_schedule(), current_index);
+      spread_rates = spreadrate.yearly_rate(simulation_year);
+      auto sr = to_array(spread_rates);
+      spread_rates_vector.push_back(sr);
+    }
   }
 
   return List::create(
-    _["infected"] = infected_vector,
-    _["exposed"] = exposed,
-    _["susceptible"] = susceptible_vector,
-    _["resistant"] = resistant_vector,
-    _["mortality"] = mortality_vector,
-    _["rates"] = spread_rates_vector,
-    _["number_infected"] = number_infected,
-    _["area_infected"] = area_infected,
-    _["total_hosts"] = total_host_vector,
-    _["propogules"] = dispersers_vector
-  );
-  
+      _["infected"] = infected_vector, _["exposed"] = exposed,
+      _["susceptible"] = susceptible_vector, _["resistant"] = resistant_vector,
+      _["mortality"] = mortality_vector, _["rates"] = spread_rates_vector,
```

Comment:
I would prefer each export variable to be on a separate line, it makes it easier to glance and see what we are exporting when we want to add in new functionality like the quarantine.